src/plugins/discover/public/application/apps/main/components/top_nav/discover_topnav.tsx

```diff
@@ -53,29 +58,44 @@ export const DiscoverTopNav = ({
     [history, resetSavedSearch, savedSearch.id]
   );
 
-  const topNavMenu = useMemo(
-    () =>
-      getTopNavLinks({
-        indexPattern,
-        navigateTo,
-        savedSearch,
-        services,
-        state: stateContainer,
-        onOpenInspector,
-        searchSource,
-        onOpenSavedSearch,
-      }),
-    [
+  const topNavMenu = useMemo(() => {
+    const links = getTopNavLinks({
       indexPattern,
       navigateTo,
       savedSearch,
       services,
-      stateContainer,
+      state: stateContainer,
       onOpenInspector,
       searchSource,
       onOpenSavedSearch,
-    ]
-  );
+    });
+
+    return [
+      {
+        id: 'labs',
+        label: i18n.translate('discover.localMenu.labs', {
+          defaultMessage: 'Labs',
```

Comment:
You might consider adding an override count:

https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/components/workpad_header/labs_control/labs_control.tsx#L45

Reply:
Sound good, but I think I can't add such a batch to the TopNavMenu? https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/navigation/public/types.ts#L18